### PR TITLE
Use CI matrix for our purposes

### LINKED ISSUE #254: No coverage data is *not* sent to Coveralls
last submission was in April

### PR DESCRIPTION
**Pull request by szepeviktor**
2nd run! 🍏 

Failed in #255

Fixes #254

### CODE REVIEW

File: .github/workflows/integrate.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    branches:
+      - "master"
+
+jobs:
+  unit_tests:
+    strategy:
+      matrix:
+        operating-system: [ "ubuntu-latest" ]
+        php-version: [ "7.4" ]
+        magento-version: [ "2.3.7-p2", "2.4.0", "2.4.1", "2.4.2", "2.4.2-p1", "2.4.2-p2", "2.4.3", "2.4.3-p1" ]
```
Comment (szepeviktor) [inline .github/workflows/integrate.yml:14]: The key was that `magento-version`-s had to be added to the base matrix.

**Comment by szepeviktor** [comment]
Now it is very lovely ❤️ 

kép

**Comment by shochdoerfer** [comment]
Thank you for opening this PR. Sadly, you do you mention why this PR is important or what problems it does fix. Currently, I only see a huge change in the CI configuration and I am not willing to merge that without an in-depth explanation of the why.

As I mentioned to you in #66 "I need consistency among our project setups and don't want to deal with custom changes just because someone has a different preference for certain things.". The current CI setup "works for me"(tm) and I don't see the need to change that now quickly. At some point, I will but this will then take more planning into account.

I would really, really appreciate if you would open an issue first to discuss things before wasting your time on a PR that I won't merge.

**Comment by szepeviktor** [comment]
> an in-depth explanation of the why

All right, all right! I suspect you are a _"What's the value?"_ businessman.

I had nothing to do on Sunday an I've tries to push my complete CI knowledge into this project.
It was big FUN! I've learned about GHA matrix.

:)

**Pull request closed without merging**